oak_runtime/src/node/logger.rs

```diff
@@ -47,51 +47,66 @@ impl super::Node for LogNode {
         handle: oak_abi::Handle,
         _notify_receiver: oneshot::Receiver<()>,
     ) {
-        loop {
-            // An error indicates the Runtime is terminating. We ignore it here and keep trying to
-            // read in case a Wasm Node wants to emit remaining messages. We will return
-            // once the channel is closed.
-            //
-            // TODO(#1100): Investigate option for improving code clarity and not busy-waiting
-            // during runtime termination.
-            let _ = runtime.wait_on_channels(&[handle]);
+        'wait: loop {
+            let result = runtime.wait_on_channels(&[handle]);
```

Comment:
Would this logic read better if it used a more strongly typed `Receiver<LogMessage>` or similar? In fact perhaps in the future we should still define a gRPC interface for this node, even if it is only a single method.

Reply:
I've added another commit that moves to using a `Receiver<LogMessage>`.  I'm slightly in two minds, because this loses the ability to dump any pending log messages on Runtime termination (which ISTR finding helpful in the past), but I think the clearer layout is worth it.

Reply:
Thanks for trying it out, I do strongly prefer the new version with `Receiver`; I don't remember the internal impl details though, could it not be made to consume subsequent messages even after the runtime terminates, if we ignore the appropriate errors?

Reply:
[`ReceiverExt::receive`](https://github.com/project-oak/oak/blob/main/oak_runtime/src/io.rs#L44) would need a re-work to make that possible, which I don't think is worth it.